# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/cracklib/crack_h.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./cracklib/crack.h.orig	2005-12-14 23:03:24.000000000 +0100
+++ ./cracklib/crack.h	2005-12-14 23:03:17.000000000 +0100
@@ -0,0 +1,42 @@
+#ifndef _CRACK_H
+#define _CRACK_H
+
+/*
+  copyright:
+    Copyright (C) 1998, 1999 Jean Pierre LeJacq <jplejacq@quoininc.com>
+
+    Distributed under the GNU GENERAL PUBLIC LICENSE.
+
+  description:
+    cracklib - a pro-active password sanity library.
+
+  ROCK Linux Note:
+    this header file is copied from the debian cracklib2_2.7-16.diff.gz
+*/
+
+
+static const char CRACKLIB_DICTPATH[] = "/var/cache/cracklib/cracklib_dict";
+/*
+  Directory and prefix of the password dictionary database used by
+  several of the utilities provided with the cracklib package.  None
+  of the functions in libcrack.a actually use this.  If you want to
+  use the same database as these utilities, use this constant as the
+  second argument to FascistCheck().
+*/
+
+
+extern char const* FascistCheck(char const password[], char const dictpath[]);
+/*
+  inputs:
+    password is the user chosen potential password.
+
+    dictpath is the full path name + filename prefix of the cracklib
+    dictionary database, which is hardcoded to CRACKLIB_DICTPATH in
+    several of the utility programs that come with cracklib.
+  return:
+    The NULL pointer for a good password, or a pointer to a diagnostic
+    string if it is a bad password.
+*/
+
+
+#endif
